lint/golangci-lint.bzl

```diff
@@ -0,0 +1,126 @@
+"""API for declaring a golangci-lint lint aspect that visits go_library, go_test, and go_binary rules.
+
+```
+load("@aspect_rules_lint//lint:golangci-lint.bzl", "golangci_lint_aspect")
+
+golangci_lint = golangci_lint_aspect(
+    binary = "@@//tools:golangci-lint",
+    config = "@@//:.golangci.yaml",
+)
+```
+"""
+
+load("@bazel_skylib//rules:native_binary.bzl", "native_binary")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@rules_go//go:def.bzl", "go_context")
```

Comment:
Yeah it's unfortunate that this module ends up with dependencies on every language, but it already happened with rules_js and I expect providers will force us to do it more.

Now that bzlmod is enabled by default in the LTS Bazel version, I'm less concerned about the impacts on users.